modules/openid-federation-common/src/jsMain/kotlin/com/sphereon/oid/fed/common/jwt/JoseJwt.js.kt

```diff
@@ -0,0 +1,48 @@
+package com.sphereon.oid.fed.common.jwt
+
+@JsModule("jose")
+@JsNonModule
+external object Jose {
+    class SignJWT {
+        constructor(payload: dynamic) {
+            definedExternally
+        }
+        fun setProtectedHeader(protectedHeader: dynamic): SignJWT {
+            definedExternally
+        }
+        fun sign(key: Any?, signOptions: Any?): String {
+            definedExternally
+        }
+    }
+    fun generateKeyPair(alg: String, options: dynamic = definedExternally): dynamic
+    fun jwtVerify(jwt: String, key: Any, options: dynamic = definedExternally): dynamic
+}
+
+@JsModule("uuid")
+@JsNonModule
+external object Uuid {
+    fun v4(): String
+}
+
+@ExperimentalJsExport
+@JsExport
+actual fun sign(
+    payload: String,
+    opts: Map<String, Any>
+): String {
+    val privateKey = opts["privateKey"] ?: throw IllegalArgumentException("JWK private key is required")
+    val header = opts["jwtHeader"] as String? ?: "{\"typ\":\"JWT\",\"alg\":\"RS256\",\"kid\":\"${Uuid.v4()}\"}"
```

Comment:
You should not just create a header, let alone use a random kid value.
Although "kid" is officially not defined, normally they are the JWK thumbprint in case a JWK is used, or a DID VM in case DIDs are used.
So please make a header a first class citizen like the payload. Require it to be passed in. 2nd don't create random kids.

Reply:
I've made header the second parameter of the sign function, the key should be passed in for both js and jvm now.